src/compile/scale.ts

```diff
@@ -65,31 +65,28 @@ export function type(fieldDef: FieldDef, channel: Channel, mark: Mark): string {
         // Also, if we support color ramp, this should be ordinal too.
         return 'time'; // time has order, so use interpolated ordinal color scale.
       }
-      if (contains([ROW, COLUMN, SHAPE], channel)) {
-        return 'ordinal';
-      }
+
       if (fieldDef.scale.type !== undefined) {
         return fieldDef.scale.type;
       }
 
+      // Returns ordinal if
+      // (1) the channel is ROW, COLUMN, or SHAPE, or
+      // (2) uses a BAR or TICK mark
+      // because in that case we should not use a time scale.
+      if (contains([ROW, COLUMN, SHAPE], channel) || contains([BAR, TICK], mark)) {
```

Comment:
So I think this really depends on how we think about it.

If we don't assume that users are design-savvy.   Bar/Tick should use ordinal only if the cardinality of the dimension is low.  Otherwise, use linear/temporal.  Since we don't know cardinality, I am leaning toward making it consistent with other marks. 

However, if we assume that users are **often** design-savvy and they would only use bar/tick for this if they know that the cardinality would make sense.  In this case, defaulting BAR and TICK might make sense for temporal with `timeUnit`.  I think that for quantitative dimension like in QxQ if we make one dimension ordinal, but not another that's super weird.  (I don't know about TxT -- would people ever make that?)